test-src/manageamiset_test.js

```diff
@@ -0,0 +1,107 @@
+var slugid = require('slugid');
+var assume = require('assume');
+var debug = require('debug')('test');
+var _ = require('lodash');
+var mock = require('./mock-amisets');
+var base = require('taskcluster-base');
+
+var makeAmiSet = mock.makeAmiSet;
+
+var main = require('../lib/main');
+var helper = require('./helper');
+
+describe('AMI Set api', () => {
+
+  var amiSetDefinition = makeAmiSet();
+  var amiSetChanged = _.clone(amiSetDefinition);
+  amiSetChanged.amis.push({
+    region: 'us-east-1',
+    hvm: 'ami-0123',
+    pv: 'ami-0345',
+  });
+
+  var id = slugid.nice();
+  var lastModified;
+
+  let amiSet;
+
+  let client;
+
+  before(async () => {
+    amiSet = await main('AmiSet', {process: 'AmiSet', profile: 'test'});
+
+    client = helper.getClient();
+  });
+
+  beforeEach(async () => {
+    await main('tableCleaner', {process: 'tableCleaner', profile: 'test'});
+  });
+
+  it('should be able to create an AMI set (idempotent)', async () => {
+    debug('### Create AMI Set');
+    await client.createAmiSet(id, amiSetDefinition);
+
+    debug('### Create AMI Set (again)');
+    await client.createAmiSet(id, amiSetDefinition);
+  });
+
+  it('should be able to update an AMI Set', async () => {
+    debug('### Load amiSet');
+    await client.createAmiSet(id, amiSetDefinition);
+
+    var amiSet = await client.amiSet(id);
+
+    debug('### Update amiSet');
+    try {
+      await client.updateAmiSet(id, amiSetChanged);
+    } catch (e) {
+      console.log(JSON.stringify(e));
+      throw e;
+    }
+
+    debug('### Load amiSet (again)');
+    amiSet = await client.amiSet(id);
+    lastModified = amiSet.lastModified;
+    assume(await amiSet).to.deeply.equal({
```

Comment:
Also, i think this is way outside the scope of this project, but one thing that I like to do is compare the amiSetChanged object itself.  Doing that does require that you handle the lastModified stuff, as you do correctly below.  Another option is to use Sinon to mock out the system clock.  That's really out of scope and what you have here is fine.  I might consider using `_.default` to grab the amis object from amiSetChanged